### PR TITLE
Precise Handling of Method without Body and more restricted isolated nodes.

### PR DESCRIPTION
**Pull request by Felalolf**
- added more precise handling of method without body
- restricted selection of isolated nodes to nodes of the files provided with -i

### CODE REVIEW

File: src/main/scala/viper/gobra/util/ViperChopper.scala [hunk 1]
```diff
@@ -457,22 +458,49 @@ object ViperChopper {
...
     case class DomainType(v: vpr.DomainType) extends Vertex
     case object Always extends Vertex // if something always has to be included, then it is a dependency of Always
 
-    def toVertex(m: vpr.Member): Vertex = {
+    // the creation of the following vertices has special cases and should not happen outside of the Vertex object
+    object Method { private[Vertex] def apply(methodName: String): Method = new Method(methodName) }
+    object PredicateBody { private[Vertex] def apply(predicateName: String): PredicateBody = new PredicateBody(predicateName) }
+    /** This function is only allowed to be called in the following cases:
+      * 1) applying [[toDefVertex]] to the predicate referenced by `predicateName` returns a [[Vertex.PredicateBody]] instance.
+      * 2) The result is used as the target of a dependency.
```
Comment (jcp19) [inline src/main/scala/viper/gobra/util/ViperChopper.scala:477]: Here, it means `1) && 2)`, right?

**Review by jcp19 — APPROVED** [review]
In general, looks fine but I have some questions

File: src/main/scala/viper/gobra/util/ChopperUtil.scala [hunk 2]
```diff
@@ -40,20 +40,28 @@ object ChopperUtil {
...
-        case None => x.start.line == target.start.line
-        case Some(pos) => target.start.line <= x.start.line && x.start.line <= pos.line
-      }) && x.file.getFileName == target.file.getFileName
+    val isIsolated = config.isolate match {
+      case Some(isolatedPositions) =>
+        def hitPosition(x: SourcePosition, target: SourcePosition): Boolean = {
+          (target.end match {
+            case None => x.start.line == target.start.line
+            case Some(pos) => target.start.line <= x.start.line && x.start.line <= pos.line
+          }) && x.file.getFileName == target.file.getFileName
```
[the remark this answers is not in the document]
Comment (Felalolf) [reply src/main/scala/viper/gobra/util/ChopperUtil.scala]: I now use `isSameFile`

**Pull request merged**